packages/x-date-pickers-pro/src/PickersRangeCalendarHeader/PickersRangeCalendarHeader.tsx

```diff
@@ -76,6 +78,7 @@ const PickersRangeCalendarHeader = React.forwardRef(function PickersRangeCalenda
       nextLabel={localeText.nextMonth}
       slots={slots}
       slotProps={slotProps}
+      labelId={`${labelId}-${monthIndex}-label`}
```

Comment:
Could we just pass this ID from `DateRangeCalendar` instead of re-building it here?
So that people with a custom implementation of the header get this for free?

We would just have to move the definition of `labelId` from the `useSlotProps` to the JSX part in `DateRangeCalendar` like we do for `monthIndex`.

Reply:
Thanks for a great observation! 🙏 
IMHO, it also cleaned the code up a bit.
I also added a JSDoc for this prop and thus there are some additional proptypes & API docs changes.